## Exercícios de Aula

### 1.
Uma empresa foi acusada de discriminação de gênero por algumas queixas que surgiram na
imprensa recentemente. Uma das acusações se referia à inexistência de equidade salarial entre homens e
mulheres.
Os responsáveis alegaram que a acusação não procedia e que pagava igualmente aos homens e
mulheres que exerciam funções similares. Além disso, se houvesse diferença salarial, isso seria por conta
do ‘tempo de casa’ dos funcionários. Mas isso é algo que acontece para ambos os sexos.
Na planilha “salários” do arquivo Aula4.xlsx, uma jornalista conseguiu dados de uma amostra de
salários recebidos por homens e mulheres que atuavam como analistas júniores do departamento onde as
queixas foram iniciadas .
Faça uma análise descritiva simples (calcule as principais medidas resumo, por exemplo) para
tentar explicar se a acusação tem fundamento ou não.

In [15]:
import pandas as pd
import numpy as np
import plotly.graph_objects as go
import plotly.express as px
import plotly.figure_factory as ff
import scipy.stats as stats

In [2]:
# read a table from a excel file
df_salarios = pd.read_excel('./Aula 04.xlsx', sheet_name='Salarios')
df_salarios.head()

ID  Salário Genero  Tempo na função (meses)
0   1  2700.50      F                       29
1   2  2950.35      M                       22
2   3  2800.19      F                       30
3   4  3100.85      M                       24
4   5  3000.89      M                       23

In [3]:
df_salarios.rename(columns={'Salário': 'salario',
                   'Genero':'genero',
                   'Tempo na função (meses)':'tempo_atuacao'}, inplace=True)

df_salarios["ganho_total"] = df_salarios["salario"] * df_salarios["tempo_atuacao"]

In [4]:
df_salarios.describe()

ID      salario  tempo_atuacao    ganho_total
count  21.000000    21.000000      21.000000      21.000000
mean   11.000000  2804.241429      23.809524   68043.539524
std     6.204837   433.588459       4.632700   21090.354132
min     1.000000  2150.180000      15.000000   36553.060000
25%     6.000000  2450.250000      20.000000   50410.920000
50%    11.000000  2800.190000      24.000000   69020.470000
75%    16.000000  3100.850000      28.000000   80003.750000
max    21.000000  3700.460000      31.000000  111013.800000

In [5]:
df_salarios.groupby('genero')[['salario', 'ganho_total', 'tempo_atuacao']].mean()

salario   ganho_total  tempo_atuacao
genero                                          
F       2507.537273  60683.147273      23.727273
M       3130.616000  76139.971000      23.900000

Ao compararmos os salários de homens e mulheres, não há uma grande disparidade. Entretanto, considerando que a média de tempo de atuação de ambos genêros é respectivamente 23,9 e 23,7 anos, ao compararmos o ganho total nesse período, a disparidade é enorme - homens ganharam em média 76140 mil enquanto as mulheres apenas 60683 mil de reais. Também é possível observar essa disparidade a partir do seguinte boxplot:

In [6]:
# plot boxplot with ganho_total by genero
fig = px.box(df_salarios, x="genero", y="ganho_total", title="Ganho total por gênero")
fig.show()

### Exercício 2

Uma rede varejista realizou uma promoção recentemente na qual cupons de desconto foram
enviados aos seus clientes. Depois de um tempo em que a campanha estava no ar uma pesquisa foi
realizada na loja física.
As perguntas eram relacionadas a: quantidade de itens comprados, valor da compra, tipo de
cartão utilizado, gênero, estado civil e idade.
As respostas de uma amostra de 100 clientes foi registrada e os dados estão disponíveis na
planilha “loja de departamento” do arquivo Aula4.xlsx. Uma legenda das variáveis é apresentada na
outra aba da planilha.
Avalie quais variáveis podem ser importantes para explicar o valor gasto nas compras dos
clientes. Isto é, compare o valor gasto entre as demais variáveis para responder.

In [7]:
df_loja = pd.read_excel('./Aula 04.xlsx', sheet_name='Loja de departamento')
df_loja.head()

Cliente Tipo de consumidor  Qtde de itens  Valor gasto Marca do cartão  \
0        1            Regular              1         39.5            Amex   
1        2            Regular              1        102.4            Amex   
2        3            Regular              1         22.5    Propria loja   
3        4            Regular              5        100.4            Amex   
4        5            Regular              2         54.0      MasterCard   

      Gênero Estado civil  Idade  
0  Masculino     Solteiro     32  
1   Feminino     Solteiro     36  
2   Feminino     Solteiro     32  
3   Feminino     Solteiro     28  
4   Feminino     Solteiro     34

In [9]:
df_loja.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100 entries, 0 to 99
Data columns (total 8 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Cliente             100 non-null    int64  
 1   Tipo de consumidor  100 non-null    object 
 2   Qtde de itens       100 non-null    int64  
 3   Valor gasto         100 non-null    float64
 4   Marca do cartão     100 non-null    object 
 5   Gênero              100 non-null    object 
 6   Estado civil        100 non-null    object 
 7   Idade               100 non-null    int64  
dtypes: float64(1), int64(3), object(4)
memory usage: 6.4+ KB


In [31]:
fig = px.histogram(df_loja, "Valor gasto")
fig.update_layout(title="Histograma de gastos")
fig.show()

#### Variáveis quantitativas X quantitativas

In [46]:
fig = px.scatter(df_loja, y="Valor gasto", x="Idade", title="Valor gasto por idade")
fig.add_annotation(
    y=200,
    x=70,
    text="r = {:.2f}".format(stats.pearsonr(df_loja["Valor gasto"], df_loja["Idade"])[0]),
    xanchor="center",
    yanchor="bottom",
    showarrow=False,
)
fig.show()

In [51]:
fig = px.scatter(df_loja, y="Valor gasto", x="Qtde de itens", title="Valor gasto por idade")
fig.add_annotation(
    x=15,
    y=250,
    text="r = {:.2f}".format(stats.pearsonr(df_loja["Valor gasto"], df_loja["Qtde de itens"])[0]),
    xanchor="center",
    yanchor="bottom",
    showarrow=False,
)
fig.show()

In [32]:
df_loja.head()

Cliente Tipo de consumidor  Qtde de itens  Valor gasto Marca do cartão  \
0        1            Regular              1         39.5            Amex   
1        2            Regular              1        102.4            Amex   
2        3            Regular              1         22.5    Propria loja   
3        4            Regular              5        100.4            Amex   
4        5            Regular              2         54.0      MasterCard   

      Gênero Estado civil  Idade  
0  Masculino     Solteiro     32  
1   Feminino     Solteiro     36  
2   Feminino     Solteiro     32  
3   Feminino     Solteiro     28  
4   Feminino     Solteiro     34

#### Variáveis quantitativas X quanlitativas

In [55]:
from plotly.subplots import make_subplots

fig = make_subplots(rows=2, cols=2)
y_col = "Valor gasto"
x_cols = ["Tipo de consumidor", "Marca do cartão", "Gênero", "Estado civil"]
for i, x_cols in enumerate(x_cols):
    fig.add_trace(
        go.Box(x=df_loja[x_cols], y=df_loja[y_col], name=x_cols, boxmean=True),
        row=i // 2 + 1,
        col=i % 2 + 1
    )
fig.update_layout(title="Boxplot de gastos", width=1000, height=600)
fig.show()

#### Variáveis quanlitativas X qualitativas

In [61]:
fig = px.histogram(df_loja, x="Gênero", color="Estado civil", barmode="group")
fig.show()